packages/rest/src/providers/send.provider.ts

```diff
@@ -31,17 +32,13 @@ export class SendProvider implements Provider<Send> {
     ],
     chain: RestTags.REST_MIDDLEWARE_CHAIN,
   }),
+  {scope: BindingScope.SINGLETON},
 )
 export class SendResponseMiddlewareProvider implements Provider<Middleware> {
-  constructor(
-    @inject(RestBindings.SequenceActions.SEND)
-    protected send: Send,
-    @inject(RestBindings.SequenceActions.REJECT)
-    protected reject: Reject,
-  ) {}
-
   value(): Middleware {
     return async (ctx, next) => {
+      const send = await ctx.get(RestBindings.SequenceActions.SEND);
```

Comment:
A general question for all similar changes: after change the biding scope to singleton, why they cannot be injected in the constructor but have to use `ctx.get()` instead?

Reply:
Service locators are less maintainable but they can be more performant than dependency injection. In this case, the class is now a singleton and we cannot inject some artifacts in the constructor. Please note the `ctx` is `RequestContext` which is specific for a given request.

Reply:
> Service locators are less maintainable but they can be more performant than dependency injection.

Fair enough 👍 then the change makes sense.